metaflow/client/core.py

```diff
@@ -1648,16 +1648,17 @@ def steps(self, *tags: str) -> Iterable[Step]:
     def code(self) -> Optional[MetaflowCode]:
         """
         Returns the MetaflowCode object for this run, if present.
-
-        Not all runs save their code so this call may return None in those cases.
+        Code is packed if atleast one `Step` runs remotely, else None is returned.
 
         Returns
         -------
         MetaflowCode, optional
             Code package for this run
         """
-        if "start" in self:
-            return self["start"].task.code
+        for step in self:
```

Comment:
Just a testing note: was wondering if this can be extremely slow with very tall Flows where the remote step is at the very end, but seems to work fine possibly due to `task.code` being a cached property.

tried this out with a flow with over a 100 local tasks before the first remote one and `run.code` still returns within 3 seconds, though it did not get any faster with subsequent calls.

Reply:
I assume we do not want the run level `code` property to be cached to avoid users accessing this for in-flight flows and getting `None` cached in some cases?

Reply:
I havent thought about that and honestly dont understand it to that point. Is this something that should be explicitly handled?

Reply:
That would be slow indeed. What are your thoughts making the MetaflowTask to add the code package info to the Run metadata, since its the same for all steps. And you wouldnt have to iterate over all steps. 

I have also seen discussions about packing the code for local only runs too, then this probably not be required?